### PR TITLE
Rearrange Images issue fixed

### LINKED ISSUE #587: Error displayed when images are removed in `Rearrange images` section
Describe your issue here.

### Steps to reproduce
- go to `images to PDF`
- select few images.
- go to `rearrange images`. Select `X` for every image
- press back button on title bar
- select `Create PDF` button

### Expected behaviour
- `create PDF` button should be disabled when no images are currently selected
- The textview at bottom which shows `X images selected` should also hide

### Actual behaviour
- Snackbar "Error on creating application folder" is displayed

### Would you like to work on the issue?
Tell us if you would like to work on this issue

### PR DESCRIPTION
**Pull request by gateway2rohan**
# Description

The issue with the "Rearrange Images" section is fixed.
Now, when all the selected images are removed in the rearrange section, the textview showing the no of images gets hidden and the "create pdf" button gets disabled.

Fixes #587 
create pdf button gets disabled and the textview showing "x  image(s) selected" gets hidden
after all the images are removed from the list.

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Tue Jan 08 21:49:33 IST 2019
+#Tue Mar 05 12:57:16 IST 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.1-all.zip
```
Comment (vyankatesh24) [inline gradle/wrapper/gradle-wrapper.properties]: Don't change configuration files.
If you are required to change, exclude them from the commit.

File: build.gradle [hunk 2]
```diff
@@ -9,7 +9,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.1'
+        classpath 'com.android.tools.build:gradle:3.3.1'
```
Comment (vyankatesh24) [inline build.gradle]: Don't change configuration files.
If you are required to change, exclude them from the commit.

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
 import android.support.v7.widget.Toolbar;
+import android.util.Log;
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: Remove unused import statement.

**Comment by vyankatesh24** [comment]
Well done! @gateway2rohan 
Just make these changes.

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java [hunk 4]
```diff
@@ -80,7 +81,7 @@ public void onDownClick(int position) {
 
     @Override
     public void onRemoveClick(int position) {
-        if (mSharedPreferences.getBoolean(Constants.CHOICE_REMOVE_IMAGE, false)) {
+        if (mSharedPreferences.getBoolean(Constants.CHOICE_REMOVE_IMAGE, false))    {
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: unnecessary space. Remove it please

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java [hunk 5]
```diff
@@ -95,10 +96,10 @@ public void onRemoveClick(int position) {
                         }
                         mImages.remove(position);
                         mRearrangeImagesAdapter.positionChanged(mImages);
-
                     })
                     .show();
         }
+
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: remove this blank line.

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 6]
```diff
@@ -318,8 +318,9 @@ public void onRequestPermissionsResult(int requestCode,
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         mButtonClicked = 0;
-        if (resultCode != Activity.RESULT_OK || data == null)
+        if (resultCode != Activity.RESULT_OK || data == null) {
             return;
+        }
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: No need for the braces here. Don't touch unnecessary things please.

Comment (gateway2rohan) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: removed the unnecessary things. Should I close this pull request and make another pull request?

**Comment by gateway2rohan** [comment]
Done

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java (hunk 6 above)
Comment (srv-twry) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: No, don't open another PR. Whatever the changes are, it should be done in this PR only.

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java [hunk 7]
```diff
@@ -95,7 +95,6 @@ public void onRemoveClick(int position) {
                         }
                         mImages.remove(position);
                         mRearrangeImagesAdapter.positionChanged(mImages);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: This is also extra change

File: gradle/wrapper/gradle-wrapper.properties [hunk 8]
```diff
@@ -1,4 +1,4 @@
-#Tue Jan 08 21:49:33 IST 2019
```
Comment (Swati4star) [inline gradle/wrapper/gradle-wrapper.properties]: remove this file from commit also

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java (hunk 7 above)
Comment (gateway2rohan) [reply app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: I didn't get you. Do you mean to remove these two lines of code?

File: gradle/wrapper/gradle-wrapper.properties (hunk 8 above)
Comment (gateway2rohan) [reply gradle/wrapper/gradle-wrapper.properties]: this is done

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java (hunk 7 above)
Comment (vyankatesh24) [reply app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: No the blank line which has been removed.

File: app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java [hunk 9]
```diff
@@ -95,7 +95,6 @@ public void onRemoveClick(int position) {
...
-
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: this line you have removed

Comment (gateway2rohan) [reply app/src/main/java/swati4star/createpdf/activity/RearrangeImages.java]: oops. Fixed it.

File: gradle/wrapper/gradle-wrapper.properties (hunk 8 above)
Comment (vyankatesh24) [reply gradle/wrapper/gradle-wrapper.properties]: @gateway2rohan
remove this file from the commit.

**Comment by Swati4star** [comment]
Congrats on your first contribution to this project! :D

**Pull request merged**